code/nebula/neb.cpp

```diff
@@ -348,6 +290,26 @@ void neb2_level_init()
 }
 
 float nNf_near, nNf_density;
+
+void neb2_poof_setup() {
+	// make the total density of poofs be the average of all poofs, and each poofs density is its relative proportion compared to others
+	// this way we maintain the retail way of not affecting total density by having more poof types
+	Poof_dist_threshold = 9999.0f;
+	float Poof_density_sum_square = 0.0f;
+	float Poof_density_sum = 0.0f;
+	for (size_t i = 0; i < Poof_info.size(); i++) {
+		if (poof_is_used(i)) {
+			Poof_density_sum_square += Poof_info[i].density * Poof_info[i].density;
+			Poof_density_sum += Poof_info[i].density;
+			float dist_threshold = Poof_info[i].view_dist * (UPKEEP_DIST_MULT - 1.0f);
+			if (dist_threshold < Poof_dist_threshold)
+				Poof_dist_threshold = dist_threshold;
+		}
+	}
+	Poof_density_multiplier = Poof_density_sum_square / (Poof_density_sum * Poof_density_sum);
```

Comment:
This will divide by 0 if all poofs are disabled. In practice, this results in the value never being read, so no issue.
However, in theory, this is undefined behaviour in C++ and should, just to be safe, be prevented.